pmtiles/bucket.go

```diff
@@ -67,6 +67,49 @@ func (m mockBucket) NewRangeReaderEtag(_ context.Context, key string, offset int
 	return io.NopCloser(bytes.NewReader(bs[offset:(offset + length)])), resultEtag, nil
 }
 
+// FileBucket is a bucket backed by a directory on disk
+type FileBucket struct {
+	path string
+	file *os.File
+}
+
+func (b FileBucket) NewRangeReader(ctx context.Context, key string, offset, length int64) (io.ReadCloser, error) {
+	body, _, err := b.NewRangeReaderEtag(ctx, key, offset, length, "")
+	return body, err
+}
+
+func (b FileBucket) NewRangeReaderEtag(ctx context.Context, key string, offset, length int64, etag string) (io.ReadCloser, string, error) {
+	name := b.path + string(os.PathSeparator) + key
```

Comment:
There are some potential optimizations like caching the file handle between requests, but that seems a bit dangerous in case the os keeps the same file handle if the underlying file is moved. I also tested performance with tilesiege and am getting 17-18k qps on a 32-core machine with both this simple implementation or caching the file handle.

Reply:
I think there's a go std library function to do combining of paths that is preferred over `+`, maybe `filepath.Join` ?